domain/domain.go

```diff
@@ -169,7 +169,18 @@ func (do *Domain) loadInfoSchema(startTS uint64) (infoschema.InfoSchema, bool, i
 		return nil, false, currentSchemaVersion, nil, err
```

Comment:
Consider if the following scenario is possible:
The first read to `loadInfoSchema(10)` is actually version 9;
The snapshot call: After the 10 commit, the second read `loadInfoSchema(10)` results in version 10 (read from `infoCache`).